src/Stoa.ts

```diff
@@ -45,15 +45,30 @@ class Stoa {
             this.ledger_storage.getValidatorsAPI(height, null,
                 (rows: any[]) =>
                 {
-                    let out_put:Array<ValidatorData> = new Array<ValidatorData>();
                     if (rows.length)
                     {
                         let out_put:Array<ValidatorData> = new Array<ValidatorData>();
 
                         for (const row of rows)
                         {
-                            let preimage: IPreimage = {distance: row.distance,
-                                hash: (row.distance == 0 ? row.random_seed : '')} as IPreimage;
+                            let preimage_hash: string = row.preimage_hash;
+                            let preimage_distance: number = row.preimage_distance;
+                            let height_preimage = new Hash();
+
+                            // Hashing preImage
+                            if (row.preimage_hash &&
+                                row.preimage_distance >= (height - row.enrolled_at))
+                            {
+                                height_preimage.fromHexString(preimage_hash);
+                                let cnt: number = (height - row.enrolled_at);
+                                for (let i = 0; i < cnt; i++)
+                                {
+                                    height_preimage.hash(height_preimage.buffer.slice());
+                                }
+                            }
```

Comment:
https://github.com/bpfkorea/agora/blob/e7f8eeb55de9c914a6ba04b2c50b948b5e1ed0d4/source/agora/consensus/EnrollmentManager.d#L515-L532
I recommend you to check the condition and result value again.